fe/fe-core/src/main/java/com/starrocks/qe/GlobalVariable.java

```diff
@@ -87,6 +93,89 @@ public final class GlobalVariable {
     @VariableMgr.VarAttr(name = "performance_schema", flag = VariableMgr.READ_ONLY)
     private static boolean performanceSchema = false;
 
+    /**
+     * Query will be pending when BE is overloaded, if `queryQueueEnable` is true.
+     * <p>
+     * If the number of running queries of any BE `exceeds queryQueueConcurrencyHardLimit`,
+     * or memory usage rate of any BE exceeds `queryQueueMemUsedPctHardLimit`,
+     * the current query will be pending or failed:
+     * - if the number of pending queries in this FE exceeds `queryQueueMaxQueuedQueries`,
+     * the query will be failed.
+     * - otherwise, the query will be pending until all the BEs aren't overloaded anymore
+     * or timeout `queryQueuePendingTimeoutSecond`.
+     * <p>
+     * Every BE reports at interval the resources containing the number of running queries and memory usage rate
+     * to the FE leader. And the FE leader synchronizes the resource usage info to FE followers by RPC.
+     * <p>
+     * The queries only using schema meta will never been queued, because a MySQL client will
+     * query schema meta after the connection is established.
+     */
+    @VariableMgr.VarAttr(name = QUERY_QUEUE_ENABLE, flag = VariableMgr.GLOBAL | VariableMgr.INVISIBLE)
+    private static boolean queryQueueEnable = false;
```

Comment:
why these variables are invisible?

Reply:
Oh, you're right, they should be visible for user.